sphinx/texinputs/sphinxmanual.cls

```diff
@@ -86,38 +87,54 @@
 
 % Catch the end of the {abstract} environment, but here make sure the abstract
 % is followed by a blank page if the 'openright' option is used.
-%
+
+% Note added (Sphinx 1.4.4): the earlier code was wrong and in fact not really
+% needed as after an abstract there might be table of contents, part, or chapter
+% which all do the right \cleardoublepage if openright option. Besides, there
+% does not seem to be Sphinx interface for user to abstract.
+\if@openright
 \let\py@OldEndAbstract=\endabstract
-\renewcommand{\endabstract}{
-  \if@openright
-    \ifodd\value{page}
-      \typeout{Adding blank page after the abstract.}
-      \vfil\pagebreak
-    \fi
-  \fi
+\renewcommand{\endabstract}{%
   \py@OldEndAbstract
+  % mimic \clearpage/\cleardoublepage with added message to log
+  \clearpage
+  % LaTeX's standard is that openright has any effect only under twoside
+  \if@twoside
+     \ifodd\c@page\else
+        \typeout{Adding blank page after the abstract}%
+        \hbox{}\newpage\if@twocolumn\hbox{}\newpage\fi
+      \fi
+  \fi
 }
+\fi
 
 % This wraps the \tableofcontents macro with all the magic to get the spacing
 % right and have the right number of pages if the 'openright' option has been
 % used.  This eliminates a fair amount of crud in the individual document files.
-%
+
+% Note added (Sphinx 1.4.4): code was wrong in case of openany option. Further:
```

Comment:
I am not sure I understand, after `\tableofcontents` I see a `\phantomsection\label{index::doc}`, and indeed if I had elsewhere ``:doc:`index``` this will make a link to that destination which turns out be Chapter 1. I would have rather that destination before the table of contents. As per "Contents:" which quickstart creates in `index.rst` by default it is a nuisance in latex except if `toctree_only` is True. 

apart from that I agree with your remarks about comments, I wll remove them. I was imagining removing them for 1.5 release. But then I need to remove obsolete legacy code comments which become inadequate now. I will make extra commit.